docs/src/snap/howto/2-node-ha.md

```diff
@@ -0,0 +1,417 @@
+# 2-Node Active-Active High-Availability using Dqlite
+
+## Rationale
+
+High availability (HA) is a mandatory requirement for most production-grade
+Kubernetes deployments, usually implying three or more nodes.
+
+2-node HA clusters are sometimes preferred for cost savings and operational
+efficiency considerations. Follow this guide to learn how Canonical Kubernetes
+can achieve high availability with just two nodes while using the default
+datastore, Dqlite.
+
+Dqlite cannot achieve Raft quorum with less than three nodes. This means that
+Dqlite will not be able to replicate data and the secondaries will simply
+forward the queries to the primary node.
+
+In the event of a node failure, database recovery will require following the
+steps in the [Dqlite recovery guide].
+
+## Proposed solution
+
+Since Dqlite data replication is not available in this situation, we propose
+using synchronous block level replication through
+[Distributed Replicated Block Device] (DRBD).
+
+The cluster monitoring and failover process will be handled by Pacemaker and
+Corosync. After a node failure, the DRBD volume will be mounted on the standby
+node, allowing access to the latest Dqlite database version.
+
+Additional recovery steps are automated and invoked through Pacemaker.
+
+## Alternatives
+
+Another possible approach is to use PostgreSQL with Kine and logical
```

Comment:
Right, so PostgreSQL was mentioned as an alternative in Antonio's document and it seemed like a good idea to cover it here as well. To be honest, I didn't get to try it out but I assumed it would be possible based on the Canonical Kubernetes docs on external datastores: https://documentation.ubuntu.com/canonical-kubernetes/latest/snap/howto/external-datastore/

I can remove the "alternatives" section for now.